legacy/src/main/java/com/amazon/opendistroforelasticsearch/sql/legacy/plugin/RestSQLQueryAction.java

```diff
@@ -179,4 +197,17 @@ private void sendResponse(RestChannel channel, RestStatus status, String content
         status, "application/json; charset=UTF-8", content));
   }
 
+  private static void logAndPublishMetrics(Exception e) {
+    if (isServerError(e)) {
+      LOG.error(LogUtils.getRequestId() + " Server side error during query execution", e);
+      Metrics.getInstance().getNumericalMetric(MetricName.FAILED_REQ_COUNT_SYS).increment();
+    } else {
+      LOG.error(LogUtils.getRequestId() + " Client side error during query execution", e);
+      Metrics.getInstance().getNumericalMetric(MetricName.FAILED_REQ_COUNT_CUS).increment();
+    }
+  }
+
+  private static boolean isServerError(Exception e) {
+    return e instanceof QueryEngineException;
```

Comment:
(Assuming this check is necessary as discussed above) I think this method should be `isClientError() { return e is SyntaxCheckException or SemanticCheckException}`? Otherwise like the `SELECT DATE('0')` example in PR description, semantic error is counted as server error.

Reply:
I think it's ambiguous in the definition query engine exception now, since it consists of semantic check and expression evaluation, but we are treating semantic check exception as client error